frontend/src/Frontend/Transfer.hs

```diff
@@ -0,0 +1,235 @@
+{-# LANGUAGE TupleSections              #-}
+{-# LANGUAGE TypeApplications           #-}
+
+module Frontend.Transfer where
+
+------------------------------------------------------------------------------
+import qualified Data.Aeson as A
+import           Data.Foldable
+import           Data.Functor
+import           Data.Scientific
+import           Control.Lens
+import           Control.Monad
+import           Control.Monad.Reader
+import qualified Data.Map as M
+import           Data.Maybe hiding (mapMaybe)
+import           Data.Text (Text)
+import qualified Data.Text as T
+import           Data.Text.Lazy (fromStrict)
+import qualified Data.Text.Lazy.Encoding as T
+import           Data.Text.Read (decimal)
+import           GHCJS.DOM.Types (MonadJSM)
+import           Obelisk.Route
+import           Obelisk.Route.Frontend
+import           Reflex.Dom.Core hiding (Value)
+import           Reflex.Network
+------------------------------------------------------------------------------
+import           Chainweb.Api.ChainId
+import           ChainwebData.AccountDetail
+import           Common.Route
+import           Common.Types
+import           Common.Utils
+import           Frontend.Accounts
+import           Frontend.App
+import           Frontend.AppState
+import           Frontend.ChainwebApi
+import           Frontend.Common
+import           Frontend.Page.Block
+import           Frontend.Transactions
+------------------------------------------------------------------------------
+
+transferSearchPage
+  :: ( MonadApp r t m
+     , Prerender js t m
+     , MonadJSM (Performable m)
+     , HasJSContext (Performable m)
+     , RouteToUrl (R FrontendRoute) m
+     , SetRoute t (R FrontendRoute) m
+     , MonadIO m
+     )
+  => App AccountParams t m ()
+transferSearchPage = do
+  r <- askRoute
+  void $ networkView $ transferHelper <$> r
+
+transferHelper
+  :: ( MonadApp r t m
+     , DomBuilder t m
+     , MonadJSM (Performable m)
+     , HasJSContext (Performable m)
+     , Prerender js t m
+     , RouteToUrl (R FrontendRoute) m
+     , SetRoute t (R FrontendRoute) m
+     , MonadIO m
+     )
+  => AccountParams
+  -> App r t m ()
+transferHelper ap = transferWidget (apAccount ap) (apToken ap) (apChain ap) Nothing
+
+transferWidget
+  :: ( MonadApp r t m
+     , DomBuilder t m
+     , MonadJSM (Performable m)
+     , HasJSContext (Performable m)
+     , Prerender js t m
+     , Reflex t
+     , RouteToUrl (R FrontendRoute) m
+     , SetRoute t (R FrontendRoute) m
+     , MonadIO m
+     )
+  => Text
+  -> Text
+  -> Maybe Integer
+  -> Maybe Int
+  -> App r t m ()
+transferWidget account token chainid fromheight = do
+  (AppState n si _ _) <- ask -- TODO: add chainweb-data host to appstate record
+  let chainwebHost = ChainwebHost (netHost n) (_siChainwebVer si)
+      mkXhr lim off nextToken = transferXhr chainwebHost account token chainid fromheight lim off nextToken
+  case mkXhr Nothing Nothing Nothing of
```

Comment:
We should check whether we have a CW-D URL available for this network and replace the contents of this widget with a helpful message informing the user that this widget is not available for the network.